packages/react/src/components/Tag/Tag.tsx

```diff
@@ -172,9 +179,9 @@ const Tag = <T extends React.ElementType>({
   }
 
   if (filter) {
-    const ComponentTag = BaseComponent ?? 'div';
+    const ComponentTag = (BaseComponent as React.ElementType) ?? 'div';
     return (
-      <ComponentTag className={tagClasses} id={tagId} {...other}>
+      <ComponentTag ref={tagRef} className={tagClasses} id={tagId} {...other}>
```

Comment:
should this not be ref too now?

Reply:
Actually we can remove the ref from this one. The `filter` prop should be replace with the new variant `Dissmisible Tag` so we don't have to add the new functionality to the old `filter` tag.

Reply:
but isn't the `Dismissible Tag` experimental? so you still need it there until it is fully deprecated

Reply:
I see your point. It is something I will take to the team to talk about it. Since we have implemented in the `Dissmisible Tag` it would be a quick to implement on the old filter prop.
But for now the spec we have is only in the new Intereactive Tag.